src/bson-encode.c

```diff
@@ -113,7 +116,32 @@ static int php_phongo_is_array_or_document(zval *val TSRMLS_DC) /* {{{ */
  * will be appended as an embedded document. */
 static void php_phongo_bson_append_object(bson_t *bson, php_phongo_bson_flags_t flags, const char *key, long key_len, zval *object TSRMLS_DC) /* {{{ */
 {
-	bson_t child;
+	if (Z_TYPE_P(object) == IS_OBJECT && instanceof_function(Z_OBJCE_P(object), php_phongo_typewrapper_ce TSRMLS_CC)) {
+#if PHP_VERSION_ID >= 70000
+		zval retval;
+
+		zend_call_method_with_0_params(object, NULL, NULL, "toBSONType", &retval);
+#else
+		zval *retval;
+
+		zend_call_method_with_0_params(&object, NULL, NULL, "toBSONType", &retval);
+#endif
+
+		if (Z_ISUNDEF(retval)) {
+			/* zend_call_method() failed */
+			return;
```

Comment:
[`zend_call_method()`](https://github.com/php/php-src/blob/PHP-7.1.4/Zend/zend_interfaces.c#L34) uses `zend_error_noreturn()`. I suppose PHP 7 might throw an Error object from that, but I expect basic error reporting in PHP 5.x.

This is consistent with other handling we do for potential `zend_call_method()` failures (e.g. `bsonSerialize()`).